### PR TITLE
Update configurations and package logic to enable builds from master

### LINKED ISSUE #102: VM cannot be attached to backend pool (NIC needs to either use a Standard SKU PIP or not have a PIP attached) & backend pool reconciliation code doesn't exist
/kind bug

Fixed in #99 
ref: #92

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: v0.1.0-alpha.3
- Kubernetes version: (use `kubectl version`): 1.13.3
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04

/assign

### LINKED ISSUE #103: Store public IP DNS name in Cluster object for use in machine scripts (includes updating associated types)
/kind feature

Fixed in #99 
ref: #92

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: v0.1.0-alpha.3
- Kubernetes version: (use `kubectl version`): 1.13.3
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04

/assign

### PR DESCRIPTION
**Pull request by justaugustus**
**What this PR does / why we need it**:

Add/update/fix the following to enable capz to build from `master`:
- Disable cloud provider config
- Update deployment template (API versions + PIP parameters)
- apis: Update PublicIP type (add DNSName) and regenerate CRDs
- Wire APIServerIP.DNSName
- Update machine naming and existence logic
- lb: Update LB types, add backend pool logic
- resources: Dedupe cluster name in GetVMName
- network: Add/update NIC methods
- actuators: Update machine actuator requeues
- services: Update Azure client calls
- network: Add NSG attach logic to subnet create/update
- network: Move VM public IP creation to network package
- vm: Update scripts to set k8s version and return successfully
- cluster/examples: Add Weave CNI to addons.yaml.template
- config: Update RBAC
- network/interfaces: Improve NIC attach logic

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #102, fixes #103

**Special notes for your reviewer**:

PR enables capz to build from `master`.
There are a few things to clean up once this merges, which I will file separate issues for.

**Release note**:

```release-note
Update configurations and package logic to enable builds from `master`
```

### CODE REVIEW

**Comment by randomvariable** [comment]
the approach looks reasonable to me here.

**Comment by justaugustus** [comment]
Tariq / Nick -- this is officially ready for review.
/assign @tariq1890 @soggiest 
/hold

File: pkg/cloud/azure/services/network/interfaces.go [hunk 1]
```diff
@@ -16,14 +16,160 @@ limitations under the License.
...
+		}
+
+		// TODO: Remove debug once reconcile logic is improved.
+		klog.V(2).Info("ReconcileNICBackendPool: Never checked the state of existing IP configuration")
+		klog.V(2).Info("Setting NIC backend pool")
+		(*nic.InterfacePropertiesFormat.IPConfigurations)[0].InterfaceIPConfigurationPropertiesFormat.LoadBalancerBackendAddressPools = &backendPools
+
+		_, err = s.CreateOrUpdateNetworkInterface(s.scope.ClusterConfig.ResourceGroup, networkInterfaceName, nic)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to update NIC %q", networkInterfaceName)
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/interfaces.go]:
```suggestion
			return errors.Wrapf(err, "Failed to update NIC %s", networkInterfaceName)
```

File: pkg/cloud/azure/services/network/interfaces.go [hunk 2]
```diff
@@ -16,14 +16,160 @@ limitations under the License.
...
+
+	return nil
+}
+
+// ReconcileNICPublicIP attaches a backend address pool ID to the supplied NIC.
+func (s *Service) ReconcileNICPublicIP(networkInterfaceName string, publicIP network.PublicIPAddress) error {
+	klog.V(2).Info("Attempting to attach public IP to NIC")
+	nic, err := s.GetNetworkInterface(s.scope.ClusterConfig.ResourceGroup, networkInterfaceName)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get NIC %q", networkInterfaceName)
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/interfaces.go]:
```suggestion
		return errors.Wrapf(err, "Failed to get NIC %s", networkInterfaceName)
```

File: pkg/cloud/azure/services/network/network.go [hunk 3]
```diff
@@ -29,23 +30,29 @@ func (s *Service) ReconcileNetwork() (err error) {
 	// TODO: Refactor
 	// TODO: Fix hardcoded values.
 	// Reconcile network security group
-	networkSGFuture, err := s.CreateOrUpdateNetworkSecurityGroup(s.scope.ClusterConfig.ResourceGroup, "ClusterAPINSG", s.scope.ClusterConfig.Location)
+	_, err = s.CreateOrUpdateNetworkSecurityGroup(s.scope.ClusterConfig.ResourceGroup, SecurityGroupDefaultName)
 	if err != nil {
 		return fmt.Errorf("error creating or updating network security group: %v", err)
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/network.go]: Let's use `err.Wrapf` here.

File: pkg/cloud/azure/services/network/publicipaddress.go [hunk 4]
```diff
@@ -50,13 +60,30 @@ func (s *Service) CreateOrUpdatePublicIPAddress(resourceGroup string, IPName str
...
+func (s *Service) DeletePublicIPAddress(resourceGroup string, IPName string) (err error) {
+	future, err := s.scope.PublicIPAddresses.Delete(s.scope.Context, resourceGroup, IPName)
+
+	if err != nil {
+		return err
+	}
+
+	err = future.WaitForCompletionRef(s.scope.Context, s.scope.PublicIPAddresses.Client)
+	if err != nil {
+		return fmt.Errorf("cannot get public IP delete future response: %v", err)
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/publicipaddress.go]:
```suggestion
		return errors.Wrapf(err, "cannot get public IP delete future response")
```

File: pkg/cloud/azure/services/network/interfaces.go [hunk 5]
```diff
@@ -16,14 +16,160 @@ limitations under the License.
...
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// GetNetworkInterface returns a network interface, if it exists.
+func (s *Service) GetNetworkInterface(resourceGroup, networkInterfaceName string) (network.Interface, error) {
+	klog.V(2).Infof("Attempting to find existing network interface %q", networkInterfaceName)
+	nic, err := s.scope.Interfaces.Get(s.scope.Context, resourceGroup, networkInterfaceName, "")
+
+	if err != nil {
+		return nic, err
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/interfaces.go]:
```suggestion
		return nil, err
```

File: pkg/cloud/azure/services/network/interfaces.go [hunk 6]
```diff
@@ -16,14 +16,160 @@ limitations under the License.
...
+
+// CreateOrUpdateNetworkInterface creates a new network interface or updates an
+// existing one, using the supplied parameters.
+func (s *Service) CreateOrUpdateNetworkInterface(resourceGroup, networkInterfaceName string, params network.Interface) (nic network.Interface, err error) {
+	klog.V(2).Infof("Attempting to update existing network interface %q", networkInterfaceName)
+	future, err := s.scope.Interfaces.CreateOrUpdate(s.scope.Context, resourceGroup, networkInterfaceName, params)
+
+	err = future.WaitForCompletionRef(s.scope.Context, s.scope.Interfaces.Client)
+	if err != nil {
+		return nic, errors.Wrapf(err, "cannot get network interface create or update future response")
```
Comment (tariq1890) [inline pkg/cloud/azure/services/network/interfaces.go]:
```suggestion
		return nic, errors.Wrapf(nil, "cannot get network interface create or update future response")
```

**Pull request merged**